packages/wonder-blocks-icon/components/icon.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+import React from "react";
+import {StyleSheet} from "aphrodite";
+import Color from "@khanacademy/wonder-blocks-color";
+
+import addStyle from "../../wonder-blocks-core/util/add-style.js";
+import iconPaths from "../util/icon-paths.js";
+
+type IconName = $Keys<typeof iconPaths>;
+type IconSize = "small" | "medium" | "large" | "xlarge";
+
+type Props = {
+    /**
+     * Passed transparently to the SVG. If not provided, we assume the SVG is
+     * purely decorative and it is invisible to screenreaders.
+     */
+    "aria-label"?: string,
+    /**
+     * The color of the icon. Will default to off-black.
+     */
+    color?: string,
+    /**
+     * One of our named icons from icon-paths.js
+     */
+    name: IconName,
+    /**
+     * One of `small` (16px), `medium` (24px), `large` (48px),
+     * or `xlarge` (96px).
+     */
+    size: IconSize,
+    /**
+     * Styles that can be processed by `addStyle` — bare style objects,
+     * Aphrodite style objects, or arrays thereof.
+     */
+    style?: any,
+};
+
+const StyledSVG = addStyle("svg");
+
+const viewportPixelsForSize: {[IconSize]: number} = {
+    small: 16,
+    medium: 24,
+    large: 48,
+    xlarge: 96,
```

Comment:
Is design actually planning on providing assets at the `large` and `xlarge` sizes for some of these assets?

Reply:
This is my understanding based off of the design spec and my conversation with Elizabeth.